File: apps/android/core/models/restaurant.kt
package com.shoplift.app.core.models

import kotlinx.serialization.Serializable

@Serializable
data class BilingualString(
    val tr: String,
    val en: String
)

@Serializable
data class Restaurant(
    val id: String,
    val name: BilingualString,
    val description: BilingualString? = null,
    val logo: String? = null,
    val address: String,
    val cuisineTags: List<String>,
    val isActive: Boolean,
    val isApproved: Boolean,
    val rating: String,
    val totalOrders: Int,
    val deliveryFee: Int,
    val taxRate: String,
    val lat: String? = null,
    val lng: String? = null,
    val createdAt: String
) {
    val displayRating: Double
        get() = rating.toDoubleOrNull() ?: 0.0
    
    val deliveryFeeDisplay: String
        get() = if (deliveryFee == 0) "Free" else "$${(deliveryFee / 100.0)}"
}

@Serializable
data class MenuItem(
    val id: String,
    val restaurantId: String,
    val name: BilingualString,
    val description: BilingualString,
    val price: String,
    val category: String,
    val imageUrl: String,
    val isAvailable: Boolean,
    val modifiers: List<ModifierGroup>,
    val displayOrder: Int
) {
    val priceDisplay: Double
        get() = price.toDoubleOrNull() ?: 0.0
    
    val priceDisplayString: String
        get() = "$${String.format("%.2f", priceDisplay)}"
}

@Serializable
data class ModifierGroup(
    val id: String,
    val name: BilingualString,
    val options: List<ModifierOption>,
    val required: Boolean,
    val maxSelections: Int
)

@Serializable
data class ModifierOption(
    val name: BilingualString,
    val price: String
) {
    val priceDisplay: Double
        get() = price.toDoubleOrNull() ?: 0.0
}

File: apps/ios/core/models/restaurant.swift
import Foundation

struct BilingualString: Codable {
    let tr: String
    let en: String
}

struct Restaurant: Codable, Identifiable {
    let id: String
    let name: BilingualString
    let description: BilingualString?
    let logo: String?
    let address: String
    let cuisineTags: [String]
    let isActive: Bool
    let isApproved: Bool
    let rating: String
    let totalOrders: Int
    let deliveryFee: Int
    let taxRate: String
    let lat: String?
    let lng: String?
    let createdAt: String
}

struct MenuItem: Codable, Identifiable {
    let id: String
    let restaurantId: String
    let name: BilingualString
    let description: BilingualString
    let price: String
    let category: String
    let imageUrl: String
    let isAvailable: Bool
    let modifiers: [ModifierGroup]
    let displayOrder: Int
}

struct ModifierGroup: Codable {
    let id: String
    let name: BilingualString
    let options: [ModifierOption]
    let required: Bool
    let maxSelections: Int
}

struct ModifierOption: Codable {
    let name: BilingualString
    let price: String
}
